src/GridFS/Bucket.php

```diff
@@ -551,6 +551,7 @@ private function getRawFileDocumentForStream($stream)
         if ( ! isset ($metadata['wrapper_data']) || ! $metadata['wrapper_data'] instanceof StreamWrapper) {
             throw InvalidArgumentException::invalidType('$stream wrapper data', isset($metadata['wrapper_data']) ? $metadata['wrapper_data'] : null, 'MongoDB\Driver\GridFS\StreamWrapper');
         }
+        $metadata['wrapper_data']->stream_close();
```

Comment:
The method I suggested https://github.com/mongodb/mongo-php-library/pull/390#issuecomment-319688537 should work. Consider:

```php
<?php

require_once 'vendor/autoload.php';

$client = new MongoDB\Client;
$bucket = $client->test->selectGridFSBucket();

$stream = $bucket->openUploadStream('example.txt');
printf("size: %d\n", fstat($stream)['size']);
var_dump($bucket->getFileDocumentForStream($stream));

fwrite($stream, 'foo');
printf("\nsize: %d\n", fstat($stream)['size']);
var_dump($bucket->getFileDocumentForStream($stream));

fclose($stream);
```

This produces the following output:

```
size: 0
object(MongoDB\Model\BSONDocument)#19 (1) {
  ["storage":"ArrayObject":private]=>
  array(4) {
    ["_id"]=>
    object(MongoDB\BSON\ObjectID)#17 (1) {
      ["oid"]=>
      string(24) "5981f438da14d80ef943da32"
    }
    ["chunkSize"]=>
    int(261120)
    ["filename"]=>
    string(11) "example.txt"
    ["uploadDate"]=>
    object(MongoDB\BSON\UTCDateTime)#18 (1) {
      ["milliseconds"]=>
      string(13) "1501688888505"
    }
  }
}

size: 3
object(MongoDB\Model\BSONDocument)#16 (1) {
  ["storage":"ArrayObject":private]=>
  array(4) {
    ["_id"]=>
    object(MongoDB\BSON\ObjectID)#18 (1) {
      ["oid"]=>
      string(24) "5981f438da14d80ef943da32"
    }
    ["chunkSize"]=>
    int(261120)
    ["filename"]=>
    string(11) "example.txt"
    ["uploadDate"]=>
    object(MongoDB\BSON\UTCDateTime)#17 (1) {
      ["milliseconds"]=>
      string(13) "1501688888505"
    }
  }
}
```

Note that you cannot call `fstat()` or `Bucket::getFileDocumentForStream()` after `fclose()`. Once closed, PHP changes the resource type from "stream" to "unknown" we can no longer work with it. If you need to access the length after `fclose()`, be sure to save the stream's ID (i.e. `_id` from the `fs.files` document) with `Bucket::getFileIdForStream($stream)`. Thereafter, you can either fetch the `fs.files` document directly with `Bucket::findOne(['_id' => $id])` or use `Bucket::openDownloadStream($id)` to obtain a new readable stream resource, which you can then use with `fstat()` or `Bucket::getFileDocumentForStream()`.